lib/nodeo/Antenna.js

```diff
@@ -14,140 +25,212 @@ var  y1=0;
   }
-  y=y*y+y1*y1;
-  y=Math.sqrt(y);
+
+  y = y*y+y1*y1;
+  y = Math.sqrt(y);
   
 return y;
 }
 
+///////////////////////////////////////////////////////////////////
 
+function trapezoidalcir(x2, upper, lower, N1, phi_desired, distance, dim) {
+/* x2          -> Vector/Array
+   upper       -> Number/Scalar
+   lower       -> Number/Scalar
+   N1          -> Number/Scalar
+   phi_desired -> Number/Scalar
+   distance    -> Number/Scalar
+   dim         -> Number/Scalar
+*/
+//This function performs integration by trapezoidal rule
+  var pi = 3.141592654;
+  var h  = (upper-lower)/N1;
+  var x1 = lower;
 
+  var asdf = array_factorcir(x2, lower, phi_desired, distance, dim);
+  var y = [Math.abs(asdf*asdf*Math.sin(lower-pi/2))];
 
-function[y sllreturn bwfn]=antennafunccircular(x1,nullMatrix,phi_desired,distance)
-//This function calculates the fitness value of array 'x1' which is returned in 'y'
-global directivity
+  for (var i=1; i<N1+1; i++) {
+      x1 = x1+h;
+      asdf = array_factorcir(x2, x1, phi_desired, distance, dim);
+      y[i] = Math.abs(asdf*asdf*Math.sin(x1-pi/2));
+  }
+
+  var s=0;
+  for (i=0; i<N1+1; i++) {
+    if (i==0 || i==N1) {
+	    s=s+y[i];
+    } else {
+	    s=s+2*y[i];
+    }
+  }
 
-var pi=3.141592654;
-var dim=length(x1);
-var y=0;
-var num_nullMatrix=nullMatrix.length;
-var num1=300;
+ return (h/2)*s;
+}
 
-var phi = []
+//////////////////////////////////////////////////////////////
+/* This function calculates the fitness value of array 'x1' which is returned in 'y'
+The returned result is a  ans  object, as defined in TODO */
+function antennafunccircular(x1, nullMatrix, phi_desired, distance) {
+/* x1          -> Vector/Array
+   nullMatrix  -> Vector/Array
+   phi_desired -> Number/Scalar
+   distance    -> Number/Scalar
+*/
+  //En Matlab,  directivity  era global
+  var directivity;
+  var pi=3.141592654;
+  var dim=x1.length;
+  var y=0;
+  var num1=300;
+  var sidelobes = [];
+  var sllphi = [];
 
-//TODO Crear un vector linea de 0 a 360 con num1 elementos
-phi=linspace(0,360,num1);
-var phizero=0;
-yax = []
-yax[0]=array_factorcir(x1,(pi/180)*phi[0],phi_desired,distance,dim);
-var maxi=yax;
-var phi_ref=1;
-//for i=2:num1//This loop finds out the maximum gain 
-for(i=1;i<num1;i++){
-    yax[i]=array_factorcir(x1,(pi/180)*phi[i],phi_desired,distance,dim);
-    if (maxi<yax[i]){
-        maxi=yax[i];
-        phizero=phi[i];
-        phi_ref=i;
-    }
+  var upper_bound=180;
+  var lower_bound=180;
+
+  var maxtem=0;
+  var count=-1;
+
+  var phi = [0];
+  for (var i=1; i<num1; i++) {
+      phi.push(i*360/(num1-1));
+  }
+
+  var phizero=0;
+  var phi_ref=0; // or 1?
+
+  var yax = [];
+  var maxi;
+
+  yax[0] = array_factorcir(x1, (pi/180)*phi[0], phi_desired, distance, dim);
+  maxi=yax[0];
+
+  //This loop finds out the maximum gain 
+  for(var i=1; i<num1; i++) {
+      yax[i] = (array_factorcir(x1, (pi/180)*phi[i], phi_desired, distance, dim));

```

Comment:
Line is too long.<br>Identifier 'array_factorcir' is not in camel case.<br>Identifier 'phi_desired' is not in camel case.